### PR TITLE
Bump Flutter min to 3.0.0 and Dart to 2.17.0

### LINKED ISSUE #1162: Bump Flutter min to 3.0.0 and Dart to 2.17.0
### Description

We can unblock features such as https://github.com/getsentry/sentry-dart/issues/1053#issuecomment-1336485647 and use the new features of the language and framework.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #1162

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: file/lib/src/sentry_file.dart [hunk 1]
```diff
@@ -53,7 +53,7 @@ class SentryFile implements File {
   }
 
   @override
-  Future<File> create({bool recursive = false}) {
+  Future<File> create({bool recursive = false, bool exclusive = false}) {
```
Comment (denrase) [inline file/lib/src/sentry_file.dart]: @marandaneto Is this correct to also introduce the `exclusive` param like you did for the `createSync` method?

Comment (marandaneto) [reply file/lib/src/sentry_file.dart]: Not yet, See https://github.com/getsentry/sentry-dart/issues/1179#issuecomment-1346199802

File: min_version_test/pubspec.yaml [hunk 2]
```diff
@@ -18,10 +18,8 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: '2.12.0'
-  flutter: '2.0.0'
-  # sdk: '>=2.12.0'
-  # flutter: '>=2.0.0'
+  sdk: '>=2.17.0 <3.0.0'
+  flutter: '>=3.0.0'
```
Comment (marandaneto) [inline min_version_test/pubspec.yaml]:
```suggestion
  sdk: '2.17.0'
  flutter: '3.0.0'
```

Keep the version pinned.

**Comment by ueman** [comment]
I believe you can also fix this issue:
https://github.com/getsentry/sentry-dart/blob/fbf42afadcf2b101268f06681d0a6b951c389d61/flutter/lib/src/event_processor/flutter_enricher_event_processor.dart#L115-L118

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-dart/blob/main/dart/lib/src/utils/hash_code.dart can likely be gone as well.

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-dart/blob/3e6109b7e78b49c9469021fc3930f42d28e5dc4b/.github/workflows/min_version_test.yml#L33
https://github.com/getsentry/sentry-dart/actions/runs/3684684967/jobs/6234715367

**Pull request merged**